neural_compressor/compression/pruner/criteria.py

```diff
@@ -289,4 +292,49 @@ def on_before_optimizer_step(self):
                 mask_grad = self.modules[key].block_mask.grad.clone()
                 self.collected_grads[key].append(mask_grad)
                 self.scores[key] += mask_grad.pow(2)
+   
+
+@register_criterion('sparse_llm')
+class SparseLLMCriterion(PruningCriterion):
+    """Pruning criterion.
+    
+    The sparse_llm criterion_class is derived from PruningCriterion.
+
+    Args:
+        config: A config dict object that includes information about pruner and pruning criterion.
+        modules: A dict {"module_name": Tensor} that stores the pruning modules' weights.
+        alpha: A parameter that determines how much of the snip score is preserved from last pruning step.
+        beta: A parameter that determines how much of the snip score is updated at the current step.
     
+    Attributes:
+        scores: A dict {"module_name": Tensor} that stores the scores of pruning modules.
+    """
+    
+    def __init__(self, modules, config):
+        """Initiliaze a sparse_llm pruning criterion."""
+        super(SparseLLMCriterion, self).__init__(modules, config)
+        
+    def set_hessian_matrix(self, module, device='cpu'):
+        W = module.weight.data.clone()
+        if isinstance(module, nn.Conv2d):
+            W = W.flatten(1)
+        if isinstance(module, transformers.Conv1D):
```

Comment:
is there any difference between torch.conv1d and transfomers.Conv1D? if not, it may be better use module.__class__.__name__